### DISCUSSION TITLE
get_export_order(): should it be a class or an instance method?

### DISCUSSION THREAD

**Discussion by m000** [django-import-export/django-import-export#1906] [General]
Hello,

I am working on migrating my v3.x django-import-export code to v4.x. I noticed some changes in the newer versions that puzzled me.

I have a custom `get_resource_kwargs(self, request, *args, **kwargs)` defined in my admin class. This is used to extract some information from the `request` object and pass them to Resource.

However, starting with v4, it seems that Resources are also instantiated by the forms without using `get_resource_kwargs()` to properly fill the init arguments. E.g. https://github.com/django-import-export/django-import-export/blob/73fc502cc377d8642c4e421b93b40e5afab77841/import_export/forms.py#L146

This means that `MyResource.__init__()` needs to account for being called from two different contexts (the form, and when actually doing the export) that may, or may not have its arguments filled-in. Even if I handle this case in `__init__()`, this means that the Resource instance will still have an uncertain state, and I will need to check for proper initialization in methods later in the workflow.

I was wondering, since the Resource constructor is called by the form with no arguments, shouldn't `get_export_order()` (and perhaps also its symmetric `get_import_order()`?) be decorated as `@classmethod`s?

**Comment by matthewhegarty** [comment]
Hi 

Thanks for raising.  The changes you referred to appeared in [this PR](https://github.com/django-import-export/django-import-export/pull/1734/files).  I would prefer that the resource is not instantiated in the form, as that seems like the incorrect place to do it, and something I missed during code review.  I'll take a look and consider how we might address this.

**Reply by m000** [reply]
Great, thanks for the ack! If you need any input for your investigation, please let me know. In the meantime, I'll add some workarounds in my code.
